ocaml/xapi/xapi_vm.ml

```diff
@@ -195,6 +195,9 @@ let start ~__context ~vm ~start_paused ~force =
 		debug "Setting ha_always_run on vm=%s as true during VM.start" (Ref.string_of vm)
 	end;
 
+	(* Check to see if we're using any restricted platform kvs. This raises
+	   an exception if so *)
+	Xapi_vm_helpers.check_restricted_platform_flags ~__context vmr.API.vM_platform;
```

Comment:
We explicitly don't want to do the check on VM.resume, just VM start, as you can always flip the switch and then start your VM without loss of data (just functionality) whereas the VM would have to be hard_shutdown to make it work if we did it on resume.

Having said that, I think it's better to keep the related functions together, so I think you're right about the location of the code. I'll move it.